mapie/multi_label_classification.py

```diff
@@ -0,0 +1,795 @@
+
+        Raises
+        ------
+        ValueError
+            Raise error if the name of the method is not
+            in self.valid_methods_
+        """
+        if self.method not in self.valid_methods_:
+            raise ValueError(
+                "Invalid method. "
+                "Allowed values are" + " or ".join(self.valid_methods_)
+            )
+
+    def _check_all_labelled(self, y: NDArray) -> None:
+        """
+        Check that all observations have at least
+        one label
+
+        Parameters
+        ----------
+        y : NDArray of shape (n_samples, n_labels)
+            Labels of the observations.
+
+        Raises
+        ------
+        ValueError
+            Raise error if at least one observation
+            has no label.
+        """
+        if not (y.sum(axis=1) > 0).all():
+            raise ValueError(
+                "Invalid y. "
+                "All observations should contain at "
+                "least one label."
+            )
+
+    def _check_delta(self, delta: Optional[float]):
+        """
+        Check that delta is not ``None`` when the
+        method is RCPS and that it is between 0 and 1.
+
+        Parameters
+        ----------
+        delta : float
+            Probability with which we control the risk. The higher
+            the probability, the more conservative the algorithm will be.
+
+        Raises
+        ------
+        ValueError
+            If delta is ``None`` and method is RCSP
+        ValueError
+            If delta is not in [0, 1] and method
+            is RCPS.
+        """
+        if (self.method == "rcps") and (delta is None):
+            raise ValueError(
+                "Invalid delta. "
+                "delta cannot be None when using "
+                "RCPS method."
+            )
+        if (not isinstance(delta, float)) and (delta is not None):
+            raise ValueError(
+                "Invalid delta. "
+                f"delta must be a float, not a {type(delta)}"
+            )
+        if (delta is not None) and (
+            ((delta <= 0) or (delta >= 1)) and
+            (self.method == "rcps")
+        ):
+            raise ValueError(
+                "Invalid delta. "
+                "delta must be in ]0, 1["
+            )
+        if (self.method == "crc") and (delta is not None):
+            warnings.warn(
+                "WARNING: you are using crc method, hence "
+                + "even if the delta is not None, it won't be"
+                + "taken into account"
+            )
+
+    def _check_estimator(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        estimator: Optional[ClassifierMixin] = None,
+        _refit: Optional[bool] = False,
+        calib_size: Optional[float] = .3
+    ) -> ClassifierMixin:
+        """
+        Check the estimator value. If it is ``None``,
+        it returns a multi-output ``LogisticRegression``
+        instance if necessary.
```

Comment:
if necessary ?! It could be unnecessary?

Reply:
yes if necessary, because ohterwise the model is NOne